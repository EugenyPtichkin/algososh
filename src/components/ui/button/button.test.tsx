import { fireEvent, render, screen } from "@testing-library/react"
import { Button } from "./button"

describe('button tests', () => {
  test('button test with text', () => {
    render(<Button text="TestButton" />)
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  test('button test without text', () => {
    render(<Button />)
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  test('button test disabled', () => {
    render(<Button disabled={true} />)
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  test('button test disabled', () => {
    render(<Button isLoader={true} />)
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  test('button test clicked on', () => {
    const btnclick = jest.fn();
    render(<Button onClick={btnclick} />)
    fireEvent.click(screen.getByRole('button'));
    expect(btnclick).toHaveBeenCalled();
  });

})