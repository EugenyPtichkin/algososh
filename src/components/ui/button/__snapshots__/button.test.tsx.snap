// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button tests button test disabled 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`button tests button test disabled 2`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button tests button test with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    TestButton
  </p>
</button>
`;

exports[`button tests button test without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
